### ISSUE TITLE
data(topoWorld) creation code is archaic

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1551]
The code in `create_data/topo/create_topo.R` is terrible. It works by binary reading of a binary file, instead of by using `read.topo()`. Heck, it also gives lon going from 0 to 360, not -180 to 180 (ish), although the docs for `data(topoWorld)` state that the range is of the latter variety.

I think I will switch to the method used in `create_data/topo/create_topo_new.R`, i.e. using a downloaded netcdf file and using `read.topo`, which is after all what we expect users to do.  

However, I see that there is something a bit wonky about the result from doing this: 
```R
library(oce)
topoFile <- download.topo(west=-179.5, east=180, south=-89.5, north=90,
                          resolution=30, format="netcdf", destdir=".")
topoWorldTest <- read.topo(topoFile)
cat(vectorShow(topoWorld[["longitude"]]))
cat(vectorShow(topoWorldTest[["longitude"]]))
cat(vectorShow(topoWorld[["latitude"]]))
cat(vectorShow(topoWorldTest[["latitude"]]))
```
gives
```R
topoWorld[["longitude"]]: -179.5, -179.0, ..., 179.5, 180.0 (length 720)
topoWorldTest[["longitude"]]: -179.25, -178.75, ..., 179.25, 179.75 (length 719)
topoWorld[["latitude"]]: -89.5, -89.0, ..., 89.5, 90.0 (length 360)
topoWorldTest[["latitude"]]: -89.25, -88.75, ..., 89.25, 89.75 (length 359)
```

In other words, 

1.  we get a different number of elements
2. the locations are offset slightly

I will look into this, giving it a higher priority than #1548 because it will affect users.

The code is a bit tricky (round about `R/topo.R` lines 836 to 864), with a test on whether there is a data element named `Band1` (what the heck is that??) and also a funky rejigging of the lon and lat, to fit z (note the FIXME there, in which I reveal a lot of confusion).  I think the code is funky because the file format has changed over the years; I've seen that kind of thing before in topo data, with e.g. websites offering multiple styles of grid, one where the z is at a corner of a lonlat cell, and another where it is at the centre of the cell.

**Comment by dankelley** [comment]
@richardsc do you mind if I change `data(topoWorld)`? If the .nc files are actually giving lon and lat as I'm inferring, then I am thinking I want the `data(topoWorld)` to do the same, and I prefer being up-to-date with data agencies, as opposed to backward-compatible with old oce datasets.

For reference, the dataset I'm downloading and reading is the result of the following URI:

https://gis.ngdc.noaa.gov/cgi-bin/public/wcs/etopo1.xyz?filename=etopo1.nc&request=getcoverage&version=1.0.0&service=wcs&coverage=etopo1&CRS=EPSG:4326&format=netcdf&resx=0.500000&resy=0.500000&bbox=-179.500000,-89.500000,180.000000,90.000000

**Comment by dankelley** [comment]
I started a sandbox on this at `sandbox/1551`, and doing 'make' in that directory causes the creation of the following file. This indicates a large deviation between the existing `data(topoWorld)` and one created by a query of the etopo1 dataset (reported at 1/60 degree, but decimated by the web query to 1/2 degree).

They are quite different, but in small features,  not large ones. The errors seem to be in spots with high slopes, so maybe this is not unexpected?

I think the question of whether to update `data(topoWorld)` is best left for a f2f meeting. I don't view this as a really high priority, because this dataset is meant for world views, and things will basically look the same there. My concern is basically that `create_data/topo/create_topoWorld.R` looks quite odd, being in terms of binary reads of a file and construction of lon and lat, instead of being in terms of `read.topo`.

![1551a](https://user-images.githubusercontent.com/99469/57572207-8c4fe380-73ed-11e9-8cf1-501f00d74002.png)

**Comment by richardsc** [comment]
Sorry! I missed the request to me to answer.

I'm fine with changing `topoWorld`. I don't think any users would notice the difference (I only use it for adding to plots -- if I want "good" bathymetry for something I usually use the `marmap` package and query the NOAA database directly).

And a good idea to clean up the creation code.

**Comment by dankelley** [comment]
Thanks, I'll do that.

**Comment by dankelley** [comment]
I've started the work. Below are the summaries of the old and the new objects. Note that the mean diff in z is under 10m, which isn't bad for a world view. The new one has mean lon and lat of 0, whereas the other means were 0.25 and 0.25.  Also, the new code is giving original names. All in all, I like it. I am also attaching a graph, illustrating that the final results will look the same on a world-scale view, which is what this dataset is intended for.

```
> summary(topoWorldOld)

Topo dataset
------------
* Source:           etopo5.dat 
* Data

                   Min.   Mean  Max. Dim.    OriginalName
    longitude [°N] -179.5 0.25  180  720     -           
    latitude [°E]  -89.5  0.25  90   360     -           
    z [m]          -9652  -1904 6401 720x360 -           

* Processing Log
    - 2016-01-24 11:42:59.113 UTC: `create 'topo' object`
    - 2016-01-24 11:42:59.113 UTC: `as.topo(longitude = longitude, latitude = latitude, z = z, filename = "etopo5.dat")`
> 
> summary(topoWorld)

Topo dataset
------------
* Source:           ./topo_180W_180E_90S_90N_30min_netcdf.nc 
* Data

                   Min.    Mean    Max.   Dim.    OriginalName
    longitude [°N] -179.75 0       179.75 720     lon         
    latitude [°E]  -89.75  0       89.75  360     lat         
    z [m]          -10471  -1892.3 6147   720x360 Band1       

* Processing Log
    - 2019-05-18 10:08:01.526 UTC: `create 'topo' object`
    - 2019-05-18 10:08:01.527 UTC: `as.topo(longitude = longitude, latitude = latitude, z = z, filename = file)`
    - 2019-05-18 10:08:01.527 UTC: `read.topo(file = topoFile, debug = 5)`
> 
```

![create_topoWorld_new](https://user-images.githubusercontent.com/99469/57970249-6e3a3400-7955-11e9-9646-b034a7aba94d.png)

**Comment by dankelley** [comment]
Done in "develop" branch, commit 132f64c9c70efdab25e80937550e3d5e264cc447

**Comment by dankelley** [comment]
I'm reopening this for an hour, to see if a rebuild of the .rda removes the following warning (from a 'check' in Rstudio).
```
─  looking to see if a ‘data/datalist’ file should be added
     NB: this package now depends on R (>= 3.5.0)
     WARNING: Added dependency on R >= 3.5.0 because serialized objects in  serialize/load version 3 cannot be read in older versions of R.  File(s) containing such objects: 'oce/data/topoWorld.rda'
```

**Comment by dankelley** [comment]
I think I see why this is happening. The `resaveRdaFiles()` call is specifying `version=2`, but the preceding `save()` call is not. I will try a rebuild/check (1/2 hour test though...) and if I am right, I will also change all the other `create_data/*/*R` files as needed.

PS. I looked in the R code to see how the check is done, but it's using a lot of code that is not exported to the user's workspace. Then I realized that a `.rda` file is gzipped, so I tried copying `topoWorld.rda` to `a.gz`, did `gunzip a.gz` and then did `head -1 a` to see
```
RDX3
```
before I change the `save()` but the required
```
RDX2
```
after I use `save("topoWorld.rda", version=2)
```

Thus, I have high confidence that I am on the right track.

PS. this comment is detailed partly so I can find it later, if required.

**Comment by dankelley** [comment]
commit 0f964553a2dd90ca79d6d024df63f4dd9422eae8
Author: dankelley <kelley.dan@gmail.com>
Date:   Tue May 21 10:17:42 2019 -0300

    Change topoWorld to be vsn 2 so old R can handle
    
    I was specifying version=2 only in the resave, not the original save. I
    also changed the coding in all the create_data/*/create*R files,
    although I am not rebuilding the rda files, given a policy of infrequent
    updates on them.
